Install more examples: `iiwa_wsg_simulation` and `kuka_simulation`

fbudin69500 opened the pull request:
Follow-up PR for #7120 which installed one of the kuka examples, but
not all (See issue #4224).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/7442)
<!-- Reviewable:end -->

jwnimmer-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> platform

+@sammy-tri for feature review (or delegate it), as area owner.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-:-KyXgwxV5pv6MebBClv8:b-38rckb)*
<!-- Sent from Reviewable.io -->

manuelli commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-:-KyXiKuj-wKGqPbkQSkZ:bnfp4nl)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-Kyq2p2ka5rspQHMPjc_:-Kyq2p2ka5rspQHMPjca:bsnv23y) (no related file):*
Oops, I dropped the ball on this one.

Examples should not be installed under `$prefix/bin`.  It is unreasonable to expect that smashing all of Drake's examples down into a single folder will be successful -- they will at best be confusing, and at worst step on each other with duplicate names.

I think the next best option is to install example binaries into `$prefix/libexec` including their full package (source-tree folder) path, similar to how `resource_tool` is currently installed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>  and I think I agree with @jwnimmer-tri on using `libexec` with package name instead of `bin`.  I had a kinda iffy feeling about this until he made that suggestion.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-:-Kyq6x4Gi6yfEz0vi8Ie:b-5hlozn)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-Kyq2p2ka5rspQHMPjc_:-KyvtlA_daCJKN8-RQZ6:b-rq5ujw) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Oops, I dropped the ball on this one.

Examples should not be installed under `$prefix/bin`.  It is unreasonable to expect that smashing all of Drake's examples down into a single folder will be successful -- they will at best be confusing, and at worst step on each other with duplicate names.

I think the next best option is to install example binaries into `$prefix/libexec` including their full package (source-tree folder) path, similar to how `resource_tool` is currently installed.
</blockquote></details>

It really doesn't matter to me, but reading the documentation about what `libexec` is intended for, it sounds like it is not suppose to be for this:
`/usr/libexec includes internal binaries that are not intended to be executed directly by users or shell scripts` (linuxfoundation website)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-Kyq2p2ka5rspQHMPjc_:-KywHLwaRh6owxnF9ImQ:b-3oyvm4) (no related file):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

It really doesn't matter to me, but reading the documentation about what `libexec` is intended for, it sounds like it is not suppose to be for this:
`/usr/libexec includes internal binaries that are not intended to be executed directly by users or shell scripts` (linuxfoundation website)
</blockquote></details>

I have no strong feelings about exactly what folder names end up as.  We just need to allow for the source folders' subdirectory names such as in `drake/examples/pendulum/lqr_simulation.cc` to come into play with the install, so that (1) the result has meaningful-to-a-user names such as `examples_pendulum_lqr_simulation` or `examples/pendulum/lqr_simulation` but not `bin/lqr_simulation`, and (2) we can re-use a drake_cc_binary name like `lqr_simulation` in several examples without stepping on each other during install.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-Kyq2p2ka5rspQHMPjc_:-Kz4EgbHaYcDsgWzdkN3:bjts3gw) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

I have no strong feelings about exactly what folder names end up as.  We just need to allow for the source folders' subdirectory names such as in `drake/examples/pendulum/lqr_simulation.cc` to come into play with the install, so that (1) the result has meaningful-to-a-user names such as `examples_pendulum_lqr_simulation` or `examples/pendulum/lqr_simulation` but not `bin/lqr_simulation`, and (2) we can re-use a drake_cc_binary name like `lqr_simulation` in several examples without stepping on each other during install.
</blockquote></details>

I moved the installation folder of the examples from `bin` to `examples/<example_sub_directory>`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-Kyq2p2ka5rspQHMPjc_:-Kz5VXjumIK6TM_wJg1k:bhvaka0) (no related file):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

I moved the installation folder of the examples from `bin` to `examples/<example_sub_directory>`
</blockquote></details>

Actually I moved the installed executables to `share/drake/examples/<example_sub_directory>`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Reviewed 1 of 1 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
@jwnimmer-tri @sammy-tri Is my update sufficient?

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-:-Kz9B5C5nOaDe67QOCI9:b-xhg40j)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
I am OK (marked resolved).  It seems weird to use `share` since the programs are a platform-dependent file, but I don't know that FHS is really our lighthouse right now anyway.

How about +@jamiesnape for review as installed-image architect, and then we can merge.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-:-Kz9Dcf9VgrxDFex8jcY:bvt6wy5)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7442#-:-Kz9Mj_kyPre1TnOb4Vm:bnfp4nl)*
<!-- Sent from Reviewable.io -->
